### PR TITLE
exclude dot files from npm

### PR DESCRIPTION
**Pull request by forceuser**

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 src/
 test/
+.*
```
Comment (ljharb) [inline .npmignore:3]: Why bother? Dotfiles are tiny (they don't hurt to include) and can be useful.

**Comment by forceuser** [comment]
The reasons are:
because dot files are for development purpose and not for end user, 
and they are not useful because we have only compiled file published to npm:

- **.gitignore** - because we can't commit anything
- **.eslintrc** - because compiled file not always corresponds to initial file code style
- **.babelrc** - same here, we have only compiled file

anyway the less files are better
the number of files affects the speed of packaging, for example when creating .zip or .jar, or any other file operations (copying, indexing), it unnoticeable with one lib but with whole `node_modules` it realy make big difference

And my initial reason: my atom editor with elsint plugin spamming error messages when i try to view compiled file because of  `.eslintrc` dependent on `airbnb-base` and i don't have `airbnb-base` installed in my project

**Comment by ljharb** [comment]
`npm install foo && npm explore foo && npm install && npm test` should always work - as such, any dotfiles related to `npm test` are required. Similarly, it's ideal for packages to publish their `.babelrc` along with their original source, so that consumers can, if they choose, source map from the compiled code to the original.

If your atom plugin is buggy, I'd recommend filing an issue on it - there's no purpose in applying eslint rules within `node_modules`.

**Comment by chicoxyzzy** [comment]
@forceuser I can understand your point but I'll let @ljharb decide if this PR should be merged or not.

I made this plugin consistent with https://github.com/airbnb/babel-plugin-dynamic-import-node and https://github.com/airbnb/babel-plugin-dynamic-import-webpack and I think we should keep all plugins consistent.

@ljharb added you as a collaborator if you want :)

**Comment by forceuser** [comment]
ok, then i close this pr

**Pull request closed without merging**